### PR TITLE
Specs for recent Module grading refactor / bug fix / features

### PR DESCRIPTION
**Pull request by sadleb**
### Description
Task: https://app.asana.com/0/1174274412967132/1200954533308122

A follow-up PR will have fixes for the floating point math bug: https://app.asana.com/0/1174274412967132/1201022023697611

Core PR that these specs are for:
* https://github.com/bebraven/platform/pull/815

Related PRs:
* https://github.com/bebraven/platform/pull/837
* https://github.com/bebraven/platform/pull/838
* https://github.com/bebraven/platform/pull/839
* https://github.com/bebraven/platform/pull/840
* https://github.com/bebraven/platform/pull/842
* https://github.com/bebraven/platform/pull/843
* https://github.com/bebraven/platform/pull/849

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/services/compute_rise360_module_grade.rb [hunk 1]
```diff
@@ -145,6 +152,13 @@ def total_points_display
       "#{points_awarded.round(1)} / #{Rise360Module::POINTS_POSSIBLE.to_f.round(1)}"
     end
 
+    def ==(other)
+      return self.total_grade == other.total_grade &&
+             self.engagement_grade == other.engagement_grade &&
+             self.quiz_grade == other.quiz_grade &&
+             self.on_time_grade == other.on_time_grade &&
+             self.completed_at == other.completed_at
+    end
```
Comment (sadleb) [inline app/services/compute_rise360_module_grade.rb:161]: This made writing specs easier b/c I could use `eq(some_grade_breakdown)`

File: spec/jobs/grade_rise360_module_for_user_job_spec.rb [hunk 2]
```diff
@@ -3,103 +3,26 @@
...
+      GradeRise360ModuleForUserJob.perform_now(user, lti_launch)
+    end
+
+    it 'calls GradeRise360ModuleForUser with the correct params' do
+      expect(GradeRise360ModuleForUser).to receive(:new).with(user, course_rise360_module_version)
+        .and_return(grading_service)
+      expect(grading_service).to receive(:run).once
+      run_job
+    end
+  end
```
Comment (sadleb) [inline spec/jobs/grade_rise360_module_for_user_job_spec.rb:27]: My note from this commit: 
I think we can actually get rid of this job since it's called
from AJAX, but just in case we may want to run it from a user blocking place
in the future I'll leave it and just put this spec here
that tests nothing other than whether there are syntax errors.

File: spec/services/compute_rise360_module_grade_spec.rb [hunk 3]
```diff
@@ -12,10 +12,9 @@
     rise360_module_version: rise360_module_version,
   ) }
   let(:canvas_assignment_id) { course_rise360_module_version.canvas_assignment_id }
-  let(:assignment_overrides) { [ create(:canvas_assignment_override_section,
```
Comment (sadleb) [inline spec/services/compute_rise360_module_grade_spec.rb:15]: Note: this file used to be the specs for `lib/module_grade_calculator` which we cutover to this service. The specs are more or less a direct cutover from the original ones. We also remove all the assignment override stuff since we now use submission `cached_due_dates` when grading.

File: spec/services/compute_rise360_module_grade_spec.rb [hunk 4]
```diff
@@ -27,676 +26,478 @@
...
+      before(:each) do
+        progress.each do |value|
+          Rise360ModuleInteraction.create!(
+            verb: Rise360ModuleInteraction::PROGRESSED,
+            user: user,
+            canvas_course_id: 333,
+            canvas_assignment_id: canvas_assignment_id,
+            activity_id: activity_id,
+            progress: value,
+            new: value != maximum, # Maximum value has new: false
```
Comment (rshipp) [inline spec/services/compute_rise360_module_grade_spec.rb:309]: this seems backwards, what am I missing? wouldn't the highest progress be new and the older ones not-new?

File: spec/services/grade_rise360_modules_spec.rb [hunk 5]
```diff
@@ -215,310 +215,244 @@
...
+        canvas_course_id: course.canvas_course_id,
+        user: user_with_new_interactions,
+        canvas_assignment_id: module_with_interactions.canvas_assignment_id,
+        progress: 50,
+      )
+    }
+    let(:grade_breakdown_zero) { ComputeRise360ModuleGrade::ComputedGradeBreakdown.new(0,0,0,0,nil) }
+    let(:grade_breakdown_partial) { ComputeRise360ModuleGrade::ComputedGradeBreakdown.new(50,50,0,0,nil) }
+    let(:grade_breakdown_completed_on_time) { ComputeRise360ModuleGrade::ComputedGradeBreakdown.new(70,100,0,100,Time.now.utc.iso8601) }
+    let(:grade_breakdown_completed_late) { ComputeRise360ModuleGrade::ComputedGradeBreakdown.new(50,100,0,0,Time.now.utc.iso8601) }
```
Comment (rshipp) [inline spec/services/grade_rise360_modules_spec.rb:263]: i'm guessing it won't matter, but any reason to pass these datetimes in as strings instead of time objects?

**Review by rshipp — APPROVED** [review]
i left a couple questions but this seems fine. i didn't look for missed edge cases or anything, just that the tests make sense. i like how clean these specs look

File: spec/services/compute_rise360_module_grade_spec.rb (hunk 4 above)
Comment (sadleb) [reply spec/services/compute_rise360_module_grade_spec.rb:309]: I'm not sure. This was from the original specs. I think the intent is just to make sure that it's grabbing the maximum progress regardless of the `created_at` or `new` value.

File: spec/services/grade_rise360_modules_spec.rb (hunk 5 above)
Comment (sadleb) [reply spec/services/grade_rise360_modules_spec.rb:263]: Hmm, good point. This is only used for display at the moment but I agree that we should pass the object in and have it converted to a string internally. I'll address that in the follow-up PR with the floating point math bug fix.

Comment (sadleb) [reply spec/services/grade_rise360_modules_spec.rb:263]: Change is here:
* https://github.com/bebraven/platform/pull/854

File: spec/services/compute_rise360_module_grade_spec.rb (hunk 4 above)
Comment (rshipp) [reply spec/services/compute_rise360_module_grade_spec.rb:309]: oh no, that probably means i wrote this lol
